sxcu/sxcu.py

```diff
@@ -122,9 +132,14 @@ def upload_file(
         if self_destruct:
             data["self_destruct"] = ""
         url = join_url(self._get_api_endpoint(default_domain=False), "/files/create")
-        with open(file, "rb") as img_file:
-            files = {"file": img_file}
-            res = request_handler.post(url, files=files, data=data)
+        if isinstance(file, io.BytesIO):
+            files = {"file": file}
+        elif isinstance(file, bytes):
+            files = {"file": io.BytesIO(file)}
+        else:
+            with open(file, "rb") as img_file:
+                files = {"file": img_file}
+        res = request_handler.post(url, files=files, data=data)
```

Comment:
I think this will not work, since the file-handle is closed before passed to requests. Have you tested this change?

Reply:
you're right. I'm working on implementing your requested changes now :)